# Biolink Model schema


A high level datamodel of biological entities (genes, diseases, phenotypes, pathways, individuals, substances, etc) and their associations.

### Classes

 * [ClassDefinition(name='administrative entity', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](AdministrativeEntity.md)
    * [ClassDefinition(name='provider', singular_name=None, description='person, group, organization or project that provides a piece of information', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['agent', 'group'], mappings=['UMLSSG:ORGA'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='administrative entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Provider.md) - person, group, organization or project that provides a piece of information
 * [ClassDefinition(name='attribute', singular_name=None, description='A property or characteristic of an entity', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=['ontology class'], abstract=False, local_names=[], union_of=[], subclass_of='PATO:0000001', values_from=[], symmetric=False, defining_slots=[], slots=['subclass of'], slot_usage={}, apply_to=None, entity=False)](Attribute.md) - A property or characteristic of an entity
    * [ClassDefinition(name='onset', singular_name=None, description='The age group in which manifestations appear', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['HP:0003674'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Onset.md) - The age group in which manifestations appear
    * [ClassDefinition(name='clinical modifier', singular_name=None, description='Used to characterize and specify the phenotypic abnormalities defined in the Phenotypic abnormality subontology, with respect to severity, laterality, age of onset, and other aspects', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='HP:0012823', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ClinicalModifier.md) - Used to characterize and specify the phenotypic abnormalities defined in the Phenotypic abnormality subontology, with respect to severity, laterality, age of onset, and other aspects
    * [ClassDefinition(name='biological sex', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='PATO:0000047', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](BiologicalSex.md)
       * [ClassDefinition(name='genotypic sex', singular_name=None, description='An attribute corresponding to the genotypic sex of the individual, based upon genotypic composition of sex chromosomes.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological sex', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='PATO:0020000', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GenotypicSex.md) - An attribute corresponding to the genotypic sex of the individual, based upon genotypic composition of sex chromosomes.
       * [ClassDefinition(name='phenotypic sex', singular_name=None, description='An attribute corresponding to the phenotypic sex of the individual, based upon the reproductive organs present.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological sex', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='PATO:0001894', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PhenotypicSex.md) - An attribute corresponding to the phenotypic sex of the individual, based upon the reproductive organs present.
    * [ClassDefinition(name='frequency value', singular_name=None, description='describes the frequency of occurrence of an event or condition', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](FrequencyValue.md) - describes the frequency of occurrence of an event or condition
    * [ClassDefinition(name='zygosity', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GENO:0000133'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Zygosity.md)
    * [ClassDefinition(name='severity value', singular_name=None, description='describes the severity of a phenotypic feature or disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='attribute', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](SeverityValue.md) - describes the severity of a phenotypic feature or disease
 * [ClassDefinition(name='named thing', singular_name=None, description='a databased entity or concept/class', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:OBJC', 'WD:Q35120'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='BFO:0000001', values_from=[], symmetric=False, defining_slots=[], slots=['id', 'name', 'category', 'related to', 'node property', 'iri', 'full name', 'description', 'systematic synonym'], slot_usage={}, apply_to=None, entity=False)](NamedThing.md) - a databased entity or concept/class
    * [ClassDefinition(name='planetary entity', singular_name=None, description='Any entity or process that exists at the level of the whole planet', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='named thing', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PlanetaryEntity.md) - Any entity or process that exists at the level of the whole planet
       * [ClassDefinition(name='environmental process', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='planetary entity', mixin=False, mixins=['occurrent'], abstract=False, local_names=[], union_of=[], subclass_of='ENVO:02500000', values_from=[], symmetric=False, defining_slots=[], slots=['regulates, process to process', 'has participant', 'has input', 'precedes'], slot_usage={}, apply_to=None, entity=False)](EnvironmentalProcess.md)
       * [ClassDefinition(name='geographic location at time', singular_name=None, description='a location that can be described in lat/long coordinates, for a particular time', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='planetary entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['latitude', 'longitude', 'timepoint'], slot_usage={}, apply_to=None, entity=False)](GeographicLocationAtTime.md) - a location that can be described in lat/long coordinates, for a particular time
       * [ClassDefinition(name='geographic location', singular_name=None, description='a location that can be described in lat/long coordinates', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:GEOG'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='planetary entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['latitude', 'longitude'], slot_usage={}, apply_to=None, entity=False)](GeographicLocation.md) - a location that can be described in lat/long coordinates
       * [ClassDefinition(name='environmental feature', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='planetary entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='ENVO:00002297', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](EnvironmentalFeature.md)
    * [ClassDefinition(name='clinical entity', singular_name=None, description='Any entity or process that exists in the clinical domain and outside the biological realm. Diseases are placed under biological entities', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='named thing', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ClinicalEntity.md) - Any entity or process that exists in the clinical domain and outside the biological realm. Diseases are placed under biological entities
       * [ClassDefinition(name='clinical intervention', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='clinical entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ClinicalIntervention.md)
       * [ClassDefinition(name='clinical trial', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='clinical entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ClinicalTrial.md)
    * [ClassDefinition(name='device', singular_name=None, description='A thing made or adapted for a particular purpose, especially a piece of mechanical or electronic equipment', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:DEVI'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='named thing', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Device.md) - A thing made or adapted for a particular purpose, especially a piece of mechanical or electronic equipment
    * [ClassDefinition(name='information content entity', singular_name=None, description='a piece of information that typically describes some piece of biology or is used as support.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['information', 'information artefact', 'information entity'], mappings=['IAO:0000030', 'UMLSSG:CONC'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='named thing', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](InformationContentEntity.md) - a piece of information that typically describes some piece of biology or is used as support.
       * [ClassDefinition(name='evidence type', singular_name=None, description='Class of evidence that supports an association', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['evidence code'], mappings=['ECO:0000000'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='information content entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['eco'], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](EvidenceType.md) - Class of evidence that supports an association
       * [ClassDefinition(name='association', singular_name=None, description='A typed association between two entities, supported by evidence', note=None, comment='This is roughly the model used by biolink and ontobio at the moment', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['OBAN:association', 'rdf:Statement', 'owl:Axiom'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='information content entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['association type', 'subject', 'negated', 'relation', 'object', 'qualifiers', 'publications', 'provided by', 'association slot'], slot_usage={}, apply_to=None, entity=False)](Association.md) - A typed association between two entities, supported by evidence
          * [ClassDefinition(name='disease to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['disease to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='disease class', note=None, comment=None, examples=[Example(value='MONDO:0017314', description='Ehlers-Danlos syndrome, vascular type')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='MONDO:0000001', values_from=['mondo', 'omim', 'orphanet', 'ncit', 'doid'], symmetric=False, multivalued=False, domain=None, range='disease', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](DiseaseToThingAssociation.md)
          * [ClassDefinition(name='cell line to disease or phenotypic feature association', singular_name=None, description='An relationship between a cell line and a disease or a phenotype, where the cell line is derived from an individual with that disease or phenotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['cell line to thing association', 'thing to disease or phenotypic feature association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['cell line to disease or phenotypic feature association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='disease or phenotypic feature', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](CellLineToDiseaseOrPhenotypicFeatureAssociation.md) - An relationship between a cell line and a disease or a phenotype, where the cell line is derived from an individual with that disease or phenotype
          * [ClassDefinition(name='sequence feature relationship', singular_name=None, description='For example, a particular exon is part of a particular transcript or gene', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GMODChado:feature_relationship'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['sequence feature relationship subject', 'sequence feature relationship object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genomic entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genomic entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](SequenceFeatureRelationship.md) - For example, a particular exon is part of a particular transcript or gene
             * [ClassDefinition(name='transcript to gene relationship', singular_name=None, description='A gene is a collection of transcripts', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='sequence feature relationship', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['transcript to gene relationship subject', 'transcript to gene relationship object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='transcript', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](TranscriptToGeneRelationship.md) - A gene is a collection of transcripts
             * [ClassDefinition(name='exon to transcript relationship', singular_name=None, description='A transcript is formed from multiple exons', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='sequence feature relationship', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['exon to transcript relationship subject', 'exon to transcript relationship object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='exon', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='transcript', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ExonToTranscriptRelationship.md) - A transcript is formed from multiple exons
             * [ClassDefinition(name='gene to gene product relationship', singular_name=None, description='A gene is transcribed and potentially translated to a gene product', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='sequence feature relationship', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['gene to gene product relationship subject', 'gene to gene product relationship object', 'gene to gene product relationship relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='has gene product', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToGeneProductRelationship.md) - A gene is transcribed and potentially translated to a gene product
          * [ClassDefinition(name='biosample to thing association', singular_name=None, description='An association between a biosample and something', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['biosample to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the biosample being described', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='biosample', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](BiosampleToThingAssociation.md) - An association between a biosample and something
          * [ClassDefinition(name='gene to phenotypic feature association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['http://bio2rdf.org/wormbase_vocabulary:Gene-Phenotype-Association'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to phenotypic feature association', 'gene to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier', 'gene to phenotypic feature association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene in which variation is correlated with the phenotypic feature', note=None, comment=None, examples=[Example(value='HGNC:2197', description='COL1A1 (Human)')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToPhenotypicFeatureAssociation.md)
          * [ClassDefinition(name='case to thing association', singular_name=None, description='An abstract association for use where the case is the subject', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['case to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the case (e.g. patient) that has the property', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='case', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](CaseToThingAssociation.md) - An abstract association for use where the case is the subject
          * [ClassDefinition(name='environment to phenotypic feature association', singular_name=None, description='Any association between an environment and a phenotypic feature, where being in the environment influences the phenotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to phenotypic feature association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier', 'environment to phenotypic feature association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='environment', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](EnvironmentToPhenotypicFeatureAssociation.md) - Any association between an environment and a phenotypic feature, where being in the environment influences the phenotype
          * [ClassDefinition(name='chemical to thing association', singular_name=None, description='An interaction between a chemical entity and another entity', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['chemical to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the chemical substance or entity that is an interactor', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='chemical substance', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ChemicalToThingAssociation.md) - An interaction between a chemical entity and another entity
          * [ClassDefinition(name='functional association', singular_name=None, description='An association between a macromolecular machine (gene, gene product or complex of gene products) and either a molecular activity, a biological process or a cellular location in which a function is executed', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['functional association subject', 'functional association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene, product or macromolecular complex that has the function associated with the GO term', note=None, comment=None, examples=[Example(value='ZFIN:ZDB-GENE-050417-357', description='twist1b')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='macromolecular machine', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='class describing the activity, process or localization of the gene product', note=None, comment=None, examples=[Example(value='GO:0016301', description='kinase activity'), Example(value='GO:0045211', description='postsynaptic membrane')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['go'], symmetric=False, multivalued=False, domain=None, range='gene ontology class', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](FunctionalAssociation.md) - An association between a macromolecular machine (gene, gene product or complex of gene products) and either a molecular activity, a biological process or a cellular location in which a function is executed
             * [ClassDefinition(name='macromolecular machine to molecular activity association', singular_name=None, description='A functional association between a macromolecular machine (gene, gene product or complex) and a molecular activity (as represented in the GO molecular function branch), where the entity carries out the activity, or contributes to its execution', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='functional association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['macromolecular machine to molecular activity association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular activity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MacromolecularMachineToMolecularActivityAssociation.md) - A functional association between a macromolecular machine (gene, gene product or complex) and a molecular activity (as represented in the GO molecular function branch), where the entity carries out the activity, or contributes to its execution
             * [ClassDefinition(name='macromolecular machine to cellular component association', singular_name=None, description='A functional association between a macromolecular machine (gene, gene product or complex) and a cellular component (as represented in the GO cellular component branch), where the entity carries out its function in the cellular component', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='functional association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['macromolecular machine to cellular component association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='cellular component', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MacromolecularMachineToCellularComponentAssociation.md) - A functional association between a macromolecular machine (gene, gene product or complex) and a cellular component (as represented in the GO cellular component branch), where the entity carries out its function in the cellular component
             * [ClassDefinition(name='gene to go term association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['functional association'], mappings=['http://bio2rdf.org/wormbase_vocabulary:Gene-GO-Association'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='functional association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['gene to go term association subject', 'gene to go term association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene, product or macromolecular complex that has the function associated with the GO term', note=None, comment=None, examples=[Example(value='ZFIN:ZDB-GENE-050417-357', description='twist1b')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='class describing the activity, process or localization of the gene product', note=None, comment=None, examples=[Example(value='GO:0016301', description='kinase activity')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['go'], symmetric=False, multivalued=False, domain=None, range='gene ontology class', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToGoTermAssociation.md)
             * [ClassDefinition(name='macromolecular machine to biological process association', singular_name=None, description='A functional association between a macromolecular machine (gene, gene product or complex) and a biological process or pathway (as represented in the GO biological process branch), where the entity carries out some part of the process, regulates it, or acts upstream of it', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='functional association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['macromolecular machine to biological process association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='biological process', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MacromolecularMachineToBiologicalProcessAssociation.md) - A functional association between a macromolecular machine (gene, gene product or complex) and a biological process or pathway (as represented in the GO biological process branch), where the entity carries out some part of the process, regulates it, or acts upstream of it
          * [ClassDefinition(name='gene to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['gene to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene that is the subject of the association', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToThingAssociation.md)
          * [ClassDefinition(name='genotype to phenotypic feature association', singular_name=None, description='Any association between one genotype and a phenotypic feature, where having the genotype confers the phenotype, either in isolation or through environment', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to phenotypic feature association', 'genotype to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier', 'genotype to phenotypic feature association relation', 'genotype to phenotypic feature association subject'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='has phenotype', inverse=None, is_class_field=False, role=None), 'subject': SlotDefinition(name='subject', singular_name=None, description='genotype that is associated with the phenotypic feature', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenotypeToPhenotypicFeatureAssociation.md) - Any association between one genotype and a phenotypic feature, where having the genotype confers the phenotype, either in isolation or through environment
          * [ClassDefinition(name='molecular interaction', singular_name=None, description='An interaction at the molecular level between two physical entities', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['molecular interaction subject', 'molecular interaction relation', 'molecular interaction object', 'molecular interaction interacting molecules category'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='interaction relationship type', note=None, comment=None, examples=[Example(value='RO:0002447', description='the subject molecular phosphorylates the object molecule')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['ro', 'mi'], symmetric=False, multivalued=False, domain=None, range='molecularly interacts with', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of='molecularly interacts with', inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'interacting molecules category': SlotDefinition(name='interacting molecules category', singular_name=None, description=None, note=None, comment=None, examples=[Example(value='MI:1048', description='smallmolecule-protein')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='MI:1046', values_from=[], symmetric=False, multivalued=False, domain=None, range='ontology class', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MolecularInteraction.md) - An interaction at the molecular level between two physical entities
          * [ClassDefinition(name='variant to phenotypic feature association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['variant to thing association', 'entity to phenotypic feature association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier', 'variant to phenotypic feature association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='a sequence variant in which the allele state is associated in some way with the phenotype state', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](VariantToPhenotypicFeatureAssociation.md)
          * [ClassDefinition(name='chemical to gene association', singular_name=None, description='An interaction between a chemical entity and a gene or gene product', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001257'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['chemical to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['chemical to gene association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='the gene or gene product that is affected by the chemical', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ChemicalToGeneAssociation.md) - An interaction between a chemical entity and a gene or gene product
          * [ClassDefinition(name='case to phenotypic feature association', singular_name=None, description='An association between a case (e.g. individual patient) and a phenotypic feature in which the individual has or has had the phenotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to phenotypic feature association', 'case to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier'], slot_usage={}, apply_to=None, entity=False)](CaseToPhenotypicFeatureAssociation.md) - An association between a case (e.g. individual patient) and a phenotypic feature in which the individual has or has had the phenotype
          * [ClassDefinition(name='genomic sequence localization', singular_name=None, description='A relationship between a sequence feature and an entity it is localized to. The reference entity may be a chromosome, chromosome region or information entity such as a contig', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['faldo:location'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['start interbase coordinate', 'end interbase coordinate', 'genome build', 'phase', 'genomic sequence localization subject', 'genomic sequence localization object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['sequence feature'], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genomic entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['reference'], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genomic entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenomicSequenceLocalization.md) - A relationship between a sequence feature and an entity it is localized to. The reference entity may be a chromosome, chromosome region or information entity such as a contig
          * [ClassDefinition(name='genotype to gene association', singular_name=None, description='Any association between a genotype and a gene. The genotype have have multiple variants in that gene or a single one. There is no assumption of cardinality', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['genotype to gene association relation', 'genotype to gene association subject', 'genotype to gene association object'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description='the relationship type used to connect genotype to gene', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'subject': SlotDefinition(name='subject', singular_name=None, description='parent genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='gene implicated in genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenotypeToGeneAssociation.md) - Any association between a genotype and a gene. The genotype have have multiple variants in that gene or a single one. There is no assumption of cardinality
          * [ClassDefinition(name='gene to expression site association', singular_name=None, description='An association between a gene and an expression site, possibly qualified by stage/timing info.', note='TBD: introduce subclasses for distinction between wild-type and experimental conditions?', comment=None, examples=[], see_also='https://github.com/monarch-initiative/ingest-artifacts/tree/master/sources/BGee', flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object', 'relation'], slots=['gene to expression site association stage qualifier', 'gene to expression site association quantifier qualifier', 'gene to expression site association subject', 'gene to expression site association object', 'gene to expression site association relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene in which variation is correlated with the phenotypic feature', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='location in which the gene is expressed', note=None, comment=None, examples=[Example(value='UBERON:0002037', description='cerebellum')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='UBERON:0001062', values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='expression relationship', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='expressed in', inverse=None, is_class_field=False, role=None), 'stage qualifier': SlotDefinition(name='stage qualifier', singular_name=None, description='stage at which the gene is expressed in the site', note=None, comment=None, examples=[Example(value='UBERON:0000069', description='larval stage')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='life stage', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'quantifier qualifier': SlotDefinition(name='quantifier qualifier', singular_name=None, description='can be used to indicate magnitude, or also ranking', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToExpressionSiteAssociation.md) - An association between a gene and an expression site, possibly qualified by stage/timing info.
          * [ClassDefinition(name='chemical to disease or phenotypic feature association', singular_name=None, description='An interaction between a chemical entity and a phenotype or disease, where the presence of the chemical gives rise to or exacerbates the phenotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:000993'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['chemical to thing association', 'thing to disease or phenotypic feature association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['chemical to disease or phenotypic feature association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='the disease or phenotype that is affected by the chemical', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='disease or phenotypic feature', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ChemicalToDiseaseOrPhenotypicFeatureAssociation.md) - An interaction between a chemical entity and a phenotype or disease, where the presence of the chemical gives rise to or exacerbates the phenotype
          * [ClassDefinition(name='variant to disease association', singular_name=None, description=None, note=None, comment='TODO decide no how to model pathogenicity', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['variant to thing association', 'entity to disease association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'variant to disease association subject', 'variant to disease association relation', 'variant to disease association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='a sequence variant in which the allele state is associated in some way with the disease state', note=None, comment=None, examples=[Example(value='ClinVar:52241', description='NM_000059.3(BRCA2):c.7007G>C (p.Arg2336Pro)')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='E.g. is pathogenic for', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='related condition', inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='a disease that is associated with that variant', note=None, comment=None, examples=[Example(value='MONDO:0016419', description='hereditary breast cancer')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](VariantToDiseaseAssociation.md)
          * [ClassDefinition(name='genotype to variant association', singular_name=None, description='Any association between a genotype and a sequence variant.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['genotype to variant association relation', 'genotype to variant association subject', 'genotype to variant association object'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description='the relationship type used to connect genotype to gene', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'subject': SlotDefinition(name='subject', singular_name=None, description='parent genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='gene implicated in genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenotypeToVariantAssociation.md) - Any association between a genotype and a sequence variant.
          * [ClassDefinition(name='sequence variant modulates treatment association', singular_name=None, description='An association between a sequence variant and a treatment or health intervention. The treatment object itself encompasses both the disease and the drug used.', note=None, comment='An alternate way to model the same information could be via a qualifier', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['sequence variant modulates treatment association subject', 'sequence variant modulates treatment association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='variant that modulates the treatment of some disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='treatment whose efficacy is modulated by the subject variant', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='treatment', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](SequenceVariantModulatesTreatmentAssociation.md) - An association between a sequence variant and a treatment or health intervention. The treatment object itself encompasses both the disease and the drug used.
          * [ClassDefinition(name='cell line to thing association', singular_name=None, description='An relationship between a cell line and another entity', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['cell line to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='cell line', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](CellLineToThingAssociation.md) - An relationship between a cell line and another entity
          * [ClassDefinition(name='gene to gene association', singular_name=None, description='abstract parent class for different kinds of gene-gene or gene product to gene product relationships. Includes homology and interaction.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['molecular or genetic interaction'], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['gene to gene association subject', 'gene to gene association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the subject gene in the association. If the relation is symmetric, subject vs object is arbitrary. We allow a gene product to stand as proxy for the gene or vice versa', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='the object gene in the association. If the relation is symmetric, subject vs object is arbitrary. We allow a gene product to stand as proxy for the gene or vice versa', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToGeneAssociation.md) - abstract parent class for different kinds of gene-gene or gene product to gene product relationships. Includes homology and interaction.
             * [ClassDefinition(name='pairwise gene or protein interaction association', singular_name=None, description='An interaction between two genes or two gene products. May be physical (e.g. protein binding) or genetic (between genes). May be symmetric (e.g. protein interaction) or directed (e.g. phosphorylation)', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene to gene association', mixin=False, mixins=['molecular interaction'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=True, defining_slots=['subject', 'object', 'relation'], slots=['pairwise gene or protein interaction association relation'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description='interaction relationship type', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['ro', 'mi'], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of='molecularly interacts with', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](PairwiseGeneOrProteinInteractionAssociation.md) - An interaction between two genes or two gene products. May be physical (e.g. protein binding) or genetic (between genes). May be symmetric (e.g. protein interaction) or directed (e.g. phosphorylation)
             * [ClassDefinition(name='gene to gene homology association', singular_name=None, description='A homology association between two genes. May be orthology (in which case the species of subject and object should differ) or paralogy (in which case the species may be the same)', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene to gene association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=True, defining_slots=['subject', 'object', 'relation'], slots=['gene to gene homology association relation'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description='homology relationship type', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of='homologous to', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToGeneHomologyAssociation.md) - A homology association between two genes. May be orthology (in which case the species of subject and object should differ) or paralogy (in which case the species may be the same)
          * [ClassDefinition(name='variant to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=True, mixins=[], abstract=True, local_names={'ga4gh': 'variant annotation'}, union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['variant to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='a sequence variant in which the allele state is associated with some other entity', note=None, comment=None, examples=[Example(value='ClinVar:38077', description='ClinVar representation of NM_000059.3(BRCA2):c.7007G>A (p.Arg2336His)'), Example(value='ClinGen:CA024716', description='chr13:g.32921033G>C (hg19) in ClinGen')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](VariantToThingAssociation.md)
          * [ClassDefinition(name='population to population association', singular_name=None, description='An association between a two populations', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['population to population association subject', 'population to population association object', 'population to population association relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the population that form the subject of the association', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='population of individual organisms', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='the population that form the object of the association', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='population of individual organisms', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='A relationship type that holds between the subject and object populations. Standard mereological relations can be used. E.g. subject part-of object, subject overlaps object. Derivation relationships can also be used', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](PopulationToPopulationAssociation.md) - An association between a two populations
          * [ClassDefinition(name='disease or phenotypic feature association to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['disease or phenotypic feature association to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='disease or phenotype', note=None, comment=None, examples=[Example(value='MONDO:0017314', description='Ehlers-Danlos syndrome, vascular type'), Example(value='MP:0013229', description='abnormal brain ventricle size')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='disease or phenotypic feature', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](DiseaseOrPhenotypicFeatureAssociationToThingAssociation.md)
             * [ClassDefinition(name='disease or phenotypic feature association to location association', singular_name=None, description='An association between either a disease or a phenotypic feature and an anatomical entity, where the disease/feature manifests in that site.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['NCIT:R100'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='disease or phenotypic feature association to thing association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['disease or phenotypic feature association to location association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='anatomical entity in which the disease or feature is found', note=None, comment=None, examples=[Example(value='UBERON:0002048', description='lung')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](DiseaseOrPhenotypicFeatureAssociationToLocationAssociation.md) - An association between either a disease or a phenotypic feature and an anatomical entity, where the disease/feature manifests in that site.
          * [ClassDefinition(name='gene to disease association', singular_name=None, description=None, note=None, comment='NCIT:R176 refers to the inverse relationship', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:000983'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to disease association', 'gene to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'gene to disease association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='gene in which variation is correlated with the disease - may be protective or causative or associative, or as a model', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneToDiseaseAssociation.md)
             * [ClassDefinition(name='gene has variant that contributes to disease association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene to disease association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object', 'relation'], slots=['sequence variant qualifier', 'gene has variant that contributes to disease association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='A gene that has a role in modeling the disease. This may be a model organism ortholog of a known disease gene, or it may be a gene whose mutants recapitulate core features of the disease.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneHasVariantThatContributesToDiseaseAssociation.md)
             * [ClassDefinition(name='gene as a model of disease association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene to disease association', mixin=False, mixins=['model to disease mixin', 'entity to disease association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object', 'relation'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'gene as a model of disease association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='A gene that has a role in modeling the disease. This may be a model organism ortholog of a known disease gene, or it may be a gene whose mutants recapitulate core features of the disease.\n  ', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GeneAsAModelOfDiseaseAssociation.md)
          * [ClassDefinition(name='disease to phenotypic feature association', singular_name=None, description='An association between a disease and a phenotypic feature in which the phenotypic feature is associated with the disease in some way', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['entity to phenotypic feature association', 'disease to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier'], slot_usage={}, apply_to=None, entity=False)](DiseaseToPhenotypicFeatureAssociation.md) - An association between a disease and a phenotypic feature in which the phenotypic feature is associated with the disease in some way
          * [ClassDefinition(name='thing to disease or phenotypic feature association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['object'], slots=['thing to disease or phenotypic feature association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='disease or phenotype', note=None, comment=None, examples=[Example(value='MONDO:0017314', description='Ehlers-Danlos syndrome, vascular type'), Example(value='MP:0013229', description='abnormal brain ventricle size')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='disease or phenotypic feature', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ThingToDiseaseOrPhenotypicFeatureAssociation.md)
          * [ClassDefinition(name='genotype to genotype part association', singular_name=None, description='Any association between one genotype and a genotypic entity that is a sub-component of it', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['genotype to genotype part association relation', 'genotype to genotype part association subject', 'genotype to genotype part association object'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='has variant part', inverse=None, is_class_field=False, role=None), 'subject': SlotDefinition(name='subject', singular_name=None, description='parent genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='child genotype', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenotypeToGenotypePartAssociation.md) - Any association between one genotype and a genotypic entity that is a sub-component of it
          * [ClassDefinition(name='variant to population association', singular_name=None, description='An association between a variant and a population, where the variant has particular frequency in the population', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['variant to thing association', 'frequency quantifier'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['frequency qualifier', 'variant to population association has count', 'variant to population association has total', 'variant to population association has quotient', 'has percentage', 'variant to population association subject', 'variant to population association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='an allele that has a certain frequency in a given population', note=None, comment=None, examples=[Example(value='NC_000017.11:g.43051071A>T', description='17:41203088 A/C in gnomad')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='the population that is observed to have the frequency', note=None, comment=None, examples=[Example(value='ANCESTRO:0010', description='African')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='population of individual organisms', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'has quotient': SlotDefinition(name='has quotient', singular_name=None, description='frequency of allele in population, expressed as a number with allele divided by number in reference population, aka allele frequency', note=None, comment=None, examples=[Example(value='0.0001666', description=None)], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'has count': SlotDefinition(name='has count', singular_name=None, description='number in object population that carry a particular allele, aka allele count', note=None, comment=None, examples=[Example(value='4', description='4 individuals in gnomad set')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'has total': SlotDefinition(name='has total', singular_name=None, description='number all populations that carry a particular allele, aka allele number', note=None, comment=None, examples=[Example(value='24014', description='24014 individuals in gnomad set')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](VariantToPopulationAssociation.md) - An association between a variant and a population, where the variant has particular frequency in the population
          * [ClassDefinition(name='genotype to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['genotype to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='genotype that is the subject of the association', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='genotype', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](GenotypeToThingAssociation.md)
          * [ClassDefinition(name='anatomical entity to anatomical entity association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['anatomical entity to anatomical entity association subject', 'anatomical entity to anatomical entity association object'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](AnatomicalEntityToAnatomicalEntityAssociation.md)
             * [ClassDefinition(name='anatomical entity to anatomical entity ontogenic association', singular_name=None, description='A relationship between two anatomical entities where the relationship is ontogenic, i.e the two entities are related by development. A number of different relationship types can be used to specify the precise nature of the relationship', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='anatomical entity to anatomical entity association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['relation'], slots=['anatomical entity to anatomical entity ontogenic association subject', 'anatomical entity to anatomical entity ontogenic association object', 'anatomical entity to anatomical entity ontogenic association relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the structure at a later time', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='the structure at an earlier time', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='develops from', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](AnatomicalEntityToAnatomicalEntityOntogenicAssociation.md) - A relationship between two anatomical entities where the relationship is ontogenic, i.e the two entities are related by development. A number of different relationship types can be used to specify the precise nature of the relationship
             * [ClassDefinition(name='anatomical entity to anatomical entity part of association', singular_name=None, description='A relationship between two anatomical entities where the relationship is mereological, i.e the two entities are related by parthood. This includes relationships between cellular components and cells, between cells and tissues, tissues and whole organisms', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='anatomical entity to anatomical entity association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['relation'], slots=['anatomical entity to anatomical entity part of association subject', 'anatomical entity to anatomical entity part of association object', 'anatomical entity to anatomical entity part of association relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='the part', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='the whole', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='anatomical entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='part of', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](AnatomicalEntityToAnatomicalEntityPartOfAssociation.md) - A relationship between two anatomical entities where the relationship is mereological, i.e the two entities are related by parthood. This includes relationships between cellular components and cells, between cells and tissues, tissues and whole organisms
          * [ClassDefinition(name='chemical to pathway association', singular_name=None, description='An interaction between a chemical entity and a biological process or pathway', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001250'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['chemical to thing association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=['chemical to pathway association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='the pathway that is affected by the chemical', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='pathway', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ChemicalToPathwayAssociation.md) - An interaction between a chemical entity and a biological process or pathway
          * [ClassDefinition(name='entity to phenotypic feature association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'sex qualifier', 'entity to phenotypic feature association description', 'entity to phenotypic feature association object'], slot_usage={'description': SlotDefinition(name='description', singular_name=None, description='A description of specific aspects of this phenotype, not otherwise covered by the phenotype ontology class', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description='phenotypic class', note=None, comment=None, examples=[Example(value='HP:0002487', description='Hyperkinesis'), Example(value='WBPhenotype:0000180', description='axon morphology variant'), Example(value='MP:0001569', description='abnormal circulating bilirubin level')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='UPHENO:0001001', values_from=['upheno', 'hp', 'mp', 'wbphenotypic feature'], symmetric=False, multivalued=False, domain=None, range='phenotypic feature', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](EntityToPhenotypicFeatureAssociation.md)
          * [ClassDefinition(name='gene regulatory relationship', singular_name=None, description='A regulatory relationship between two genes', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['gene regulatory relationship relation', 'gene regulatory relationship subject', 'gene regulatory relationship object'], slot_usage={'relation': SlotDefinition(name='relation', singular_name=None, description='the direction is always from regulator to regulated', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role='regulatory gene'), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='gene or gene product', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role='regulated gene')}, apply_to=None, entity=False)](GeneRegulatoryRelationship.md) - A regulatory relationship between two genes
          * [ClassDefinition(name='biosample to disease or phenotypic feature association', singular_name=None, description='An association between a biosample and a disease or phenotype\n  definitional: true\n  ', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=False, mixins=['biosample to thing association', 'thing to disease or phenotypic feature association'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject', 'object'], slots=[], slot_usage={}, apply_to=None, entity=False)](BiosampleToDiseaseOrPhenotypicFeatureAssociation.md) - An association between a biosample and a disease or phenotype
  definitional: true
       * [ClassDefinition(name='publication', singular_name=None, description='Any published piece of information. Can refer to a whole publication, or to a part of it (e.g. a figure, figure legend, or section highlighted by NLP). The scope is intended to be general and include information published on the web as well as journals.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['reference'], mappings=['IAO:0000311'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='information content entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Publication.md) - Any published piece of information. Can refer to a whole publication, or to a part of it (e.g. a figure, figure legend, or section highlighted by NLP). The scope is intended to be general and include information published on the web as well as journals.
       * [ClassDefinition(name='confidence level', singular_name=None, description='Level of confidence in a statement', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['CIO:0000028'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='information content entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['cio'], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ConfidenceLevel.md) - Level of confidence in a statement
    * [ClassDefinition(name='biological entity', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['WD:Q28845870'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='named thing', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['has phenotype'], slot_usage={}, apply_to=None, entity=False)](BiologicalEntity.md)
       * [ClassDefinition(name='environment', singular_name=None, description='A feature of the environment of an organism that influences one or more phenotypic features of that organism, potentially mediated by genes', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['environment', 'exposure', 'experimental condition'], mappings=['SIO:000955'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Environment.md) - A feature of the environment of an organism that influences one or more phenotypic features of that organism, potentially mediated by genes
          * [ClassDefinition(name='treatment', singular_name=None, description="A treatment is targeted at a disease or phenotype and may involve multiple drug 'exposures'", note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['medical action'], mappings=['OGMS:0000090', 'SIO:001398'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='environment', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['treatment treats', 'treatment has exposure parts'], slot_usage={'treats': SlotDefinition(name='treats', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='disease or phenotypic feature', required=True, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'has exposure parts': SlotDefinition(name='has exposure parts', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=True, domain=None, range='drug exposure', required=True, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](Treatment.md) - A treatment is targeted at a disease or phenotype and may involve multiple drug 'exposures'
          * [ClassDefinition(name='drug exposure', singular_name=None, description='A drug exposure is an intake of a particular chemical substance', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['drug intake', 'drug dose'], mappings=['ECTO:0000509', 'SIO:001005'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='environment', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['drug exposure drug'], slot_usage={'drug': SlotDefinition(name='drug', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=True, domain=None, range='chemical substance', required=True, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](DrugExposure.md) - A drug exposure is an intake of a particular chemical substance
       * [ClassDefinition(name='biological process or activity', singular_name=None, description='Either an individual molecular activity, or a collection of causally connected molecular activities', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=['GO', 'Reactome'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=['molecular activity', 'biological process'], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](BiologicalProcessOrActivity.md) - Either an individual molecular activity, or a collection of causally connected molecular activities
          * [ClassDefinition(name='molecular activity', singular_name=None, description='An execution of a molecular function carried out by a gene product or macromolecular complex.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['molecular function', 'molecular event', 'reaction'], mappings=['GO:0003674'], id_prefixes=['GO', 'Reactome'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological process or activity', mixin=False, mixins=['occurrent'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['regulates, process to process', 'has participant', 'has input', 'precedes'], slot_usage={}, apply_to=None, entity=False)](MolecularActivity.md) - An execution of a molecular function carried out by a gene product or macromolecular complex.
          * [ClassDefinition(name='biological process', singular_name=None, description='One or more causally connected executions of molecular functions', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GO:0008150', 'SIO:000006', 'WD:Q2996394'], id_prefixes=['GO', 'Reactome'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological process or activity', mixin=False, mixins=['occurrent'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['regulates, process to process', 'has participant', 'has input', 'precedes'], slot_usage={}, apply_to=None, entity=False)](BiologicalProcess.md) - One or more causally connected executions of molecular functions
             * [ClassDefinition(name='pathway', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GO:0007165', 'SIO:010526', 'PW:0000001', 'WD:Q4915012'], id_prefixes=['GO', 'Reactome'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological process', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Pathway.md)
             * [ClassDefinition(name='physiological process', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['physiology'], mappings=['UMLSSG:PHYS'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological process', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PhysiologicalProcess.md)
       * [ClassDefinition(name='disease or phenotypic feature', singular_name=None, description='Either one of a disease or an individual phenotypic feature. Some knowledge resources such as Monarch treat these as distinct, others such as MESH conflate.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['phenome'], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=['disease', 'phenotypic feature'], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['correlated with', 'has biomarker', 'treated by', 'in taxon'], slot_usage={}, apply_to=None, entity=False)](DiseaseOrPhenotypicFeature.md) - Either one of a disease or an individual phenotypic feature. Some knowledge resources such as Monarch treat these as distinct, others such as MESH conflate.
          * [ClassDefinition(name='disease', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['condition', 'disorder', 'medical condition'], mappings=['MONDO:0000001', 'WD:Q12136', 'SIO:010299', 'UMLSSG:DISO'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='disease or phenotypic feature', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Disease.md)
          * [ClassDefinition(name='phenotypic feature', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['sign', 'symptom', 'phenotype', 'trait', 'endophenotype'], mappings=['UPHENO:0001001', 'SIO:010056', 'WD:Q169872'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='disease or phenotypic feature', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PhenotypicFeature.md)
       * [ClassDefinition(name='molecular entity', singular_name=None, description='A gene, gene product, small molecule or macromolecule (including protein complex)', note=None, comment='Check SIO mapping - also mapped to chemical substance.', examples=[], see_also=None, flags=[], prefixes=[], aliases=['bioentity'], mappings=['SIO:010004', 'WD:Q43460564'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['molecularly interacts with', 'regulates, entity to entity', 'biomarker for', 'in taxon'], slot_usage={}, apply_to=None, entity=False)](MolecularEntity.md) - A gene, gene product, small molecule or macromolecule (including protein complex)
          * [ClassDefinition(name='genomic entity', singular_name=None, description='an entity that can either be directly located on a genome (gene, transcript, exon, regulatory region) or is encoded in a genome (protein)', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['sequence feature'], mappings=['SO:0000110', 'UMLSSG:GENE'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='molecular entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['has biological sequence'], slot_usage={}, apply_to=None, entity=False)](GenomicEntity.md) - an entity that can either be directly located on a genome (gene, transcript, exon, regulatory region) or is encoded in a genome (protein)
             * [ClassDefinition(name='transcript', singular_name=None, description='An RNA synthesized on a DNA or RNA template by an RNA polymerase', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SO:0000673', 'SIO:010450'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Transcript.md) - An RNA synthesized on a DNA or RNA template by an RNA polymerase
             * [ClassDefinition(name='haplotype', singular_name=None, description='A set of zero or more Alleles on a single instance of a Sequence[VMC]', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GENO:0000871', 'VMC:Haplotype'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Haplotype.md) - A set of zero or more Alleles on a single instance of a Sequence[VMC]
             * [ClassDefinition(name='macromolecular machine', singular_name=None, description='A union of gene, gene product, and macromolecular complex. These are the basic units of function in a cell. They either carry out individual biological activities, or they encode molecules which do this.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=['gene', 'gene product', 'macromolecular complex'], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['macromolecular machine name'], slot_usage={'name': SlotDefinition(name='name', singular_name=None, description='genes are typically designated by a short symbol and a full name. We map the symbol to the default display name and use an additional slot for full name', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='symbol type', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MacromolecularMachine.md) - A union of gene, gene product, and macromolecular complex. These are the basic units of function in a cell. They either carry out individual biological activities, or they encode molecules which do this.
                * [ClassDefinition(name='gene or gene product', singular_name=None, description='a union of genes or gene products. Frequently an identifier for one will be used as proxy for another', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='macromolecular machine', mixin=False, mixins=[], abstract=False, local_names=[], union_of=['gene', 'gene product'], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['in pathway with', 'in complex with', 'in cell population with', 'expressed in'], slot_usage={}, apply_to=None, entity=False)](GeneOrGeneProduct.md) - a union of genes or gene products. Frequently an identifier for one will be used as proxy for another
                   * [ClassDefinition(name='gene', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['locus'], mappings=['SO:0000704', 'SIO:010035', 'WD:Q7187', 'UMLSSG:GENE'], id_prefixes=['NCBIGene', 'ENSEMBL', 'HGNC', 'MGI', 'ZFIN', 'dictyBase', 'WB', 'SGD', 'PomBase'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene or gene product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['genetically interacts with', 'has gene product', 'gene associated with condition'], slot_usage={}, apply_to=None, entity=False)](Gene.md)
                   * [ClassDefinition(name='gene product', singular_name=None, description='The functional molecular product of a single gene. Gene products are either proteins or functional RNA molecules', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['WD:Q424689'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene or gene product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=['protein', 'RNA product'], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GeneProduct.md) - The functional molecular product of a single gene. Gene products are either proteins or functional RNA molecules
                      * [ClassDefinition(name='protein', singular_name=None, description='A gene product that is composed of a chain of amino acid sequences and is produced by ribosome-mediated translation of mRNA', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['polypeptide'], mappings=['PR:000000001', 'SIO:010043', 'WD:Q8054'], id_prefixes=['UniProtKB', 'PR', 'ENSEMBL'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Protein.md) - A gene product that is composed of a chain of amino acid sequences and is produced by ribosome-mediated translation of mRNA
                         * [ClassDefinition(name='protein isoform', singular_name=None, description='Represents a protein that is a specific isoform of the canonical or reference protein. See https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4114032/', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['proteoform'], mappings=[], id_prefixes=['UniProtKB', 'PR', 'ENSEMBL'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='protein', mixin=False, mixins=['gene product isoform'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ProteinIsoform.md) - Represents a protein that is a specific isoform of the canonical or reference protein. See https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4114032/
                      * [ClassDefinition(name='gene product isoform', singular_name=None, description='This is an abstract class that can be mixed in with different kinds of gene products to indicate that the gene product is intended to represent a specific isoform rather than a canonical or reference or generic product. The designation of canonical or reference may be arbitrary, or it may represent the superclass of all isoforms.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene product', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=True)](GeneProductIsoform.md) - This is an abstract class that can be mixed in with different kinds of gene products to indicate that the gene product is intended to represent a specific isoform rather than a canonical or reference or generic product. The designation of canonical or reference may be arbitrary, or it may represent the superclass of all isoforms.
                      * [ClassDefinition(name='RNA product', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['CHEBI:33697', 'SIO:010450', 'WD:Q11053'], id_prefixes=['RNAcentral'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='gene product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](RnaProduct.md)
                         * [ClassDefinition(name='RNA product isoform', singular_name=None, description='Represents a protein that is a specific isoform of the canonical or reference RNA', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=['RNAcentral'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='RNA product', mixin=False, mixins=['gene product isoform'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](RnaProductIsoform.md) - Represents a protein that is a specific isoform of the canonical or reference RNA
                         * [ClassDefinition(name='noncoding RNA product', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001235'], id_prefixes=['RNAcentral', 'NCBIGene', 'ENSEMBL'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='RNA product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='SO:0000655', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](NoncodingRnaProduct.md)
                            * [ClassDefinition(name='microRNA', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001397', 'WD:Q310899'], id_prefixes=['MIR'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='noncoding RNA product', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='SO:0000276', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Microrna.md)
                * [ClassDefinition(name='macromolecular complex', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:010046', 'WD:Q22325163'], id_prefixes=['IntAct', 'GO', 'PR', 'Reactome'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='macromolecular machine', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='GO:0032991', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](MacromolecularComplex.md)
             * [ClassDefinition(name='sequence variant', singular_name=None, description='An allele that varies in its sequence from what is considered the reference allele at that locus.', note=None, comment='This class is for modeling the specific state at a locus. A single dbSNP rs ID could correspond to more than one sequence variants (e.g CIViC:1252 and CIViC:1253, two distinct BRCA2 alleles for rs28897743)', examples=[], see_also=None, flags=[], prefixes=[], aliases=['allele'], mappings=['GENO:0000002', 'WD:Q15304597', 'SIO:010277', 'VMC:Allele'], id_prefixes=['ClinVar', 'WD', 'CIViC'], in_subset=[], from_schema=None, alt_descriptions=[{'description': 'An enitity that describes a single affected, endogenous allele.  These can be of any type that matches that definition', 'source': 'AGR'}, {'description': 'A contiguous change at a Location', 'source': 'VMC'}], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names={'agr': 'allele'}, union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['sequence variant has gene', 'sequence variant has biological sequence', 'sequence variant id'], slot_usage={'has gene': SlotDefinition(name='has gene', singular_name=None, description='Each allele can be associated with any number of genes', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=True, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'has biological sequence': SlotDefinition(name='has biological sequence', singular_name=None, description='The state of the sequence w.r.t a reference sequence', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'id': SlotDefinition(name='id', singular_name=None, description=None, note=None, comment=None, examples=[Example(value='ZFIN:ZDB-ALT-980203-1091', description='ti282a allele from ZFIN'), Example(value='ClinVarVariant:17681', description='NM_007294.3(BRCA1):c.2521C>T (p.Arg841Trp)')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](SequenceVariant.md) - An allele that varies in its sequence from what is considered the reference allele at that locus.
             * [ClassDefinition(name='coding sequence', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SO:0000316', 'SIO:001390'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](CodingSequence.md)
             * [ClassDefinition(name='genome', singular_name=None, description='A genome is the sum of genetic material within a cell or virion.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SO:0001026', 'SIO:000984', 'WD:Q7020'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Genome.md) - A genome is the sum of genetic material within a cell or virion.
             * [ClassDefinition(name='genotype', singular_name=None, description='An information content entity that describes a genome by specifying the total variation in genomic sequence and/or gene expression, relative to some extablished background', note=None, comment='Consider renaming as genotypic entity', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GENO:0000536', 'SIO:001079'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['has zygosity'], slot_usage={}, apply_to=None, entity=False)](Genotype.md) - An information content entity that describes a genome by specifying the total variation in genomic sequence and/or gene expression, relative to some extablished background
             * [ClassDefinition(name='exon', singular_name=None, description='A region of the transcript sequence within a gene which is not removed from the primary RNA transcript by RNA splicing', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SO:0000147', 'SIO:010445', 'WD:Q373027'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='genomic entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Exon.md) - A region of the transcript sequence within a gene which is not removed from the primary RNA transcript by RNA splicing
          * [ClassDefinition(name='gene family', singular_name=None, description='any grouping of multiple genes or gene products related by common descent', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001380', 'NCIT:C20130', 'WD:Q417841'], id_prefixes=['PANTHER'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='molecular entity', mixin=False, mixins=['gene grouping'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GeneFamily.md) - any grouping of multiple genes or gene products related by common descent
          * [ClassDefinition(name='chemical substance', singular_name=None, description='May be a chemical entity or a formulation with a chemical entity as active ingredient, or a complex material with multiple chemical entities as part', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:010004', 'UMLSSG:CHEM', 'WD:Q79529'], id_prefixes=['CHEMBL.TARGET'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='molecular entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='CHEBI:24431', values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ChemicalSubstance.md) - May be a chemical entity or a formulation with a chemical entity as active ingredient, or a complex material with multiple chemical entities as part
             * [ClassDefinition(name='metabolite', singular_name=None, description='Any intermediate or product resulting from metabolism. Includes primary and secondary metabolites.', note=None, comment='The CHEBI ID represents a role rather than a substance', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['CHEBI:25212'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='chemical substance', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Metabolite.md) - Any intermediate or product resulting from metabolism. Includes primary and secondary metabolites.
             * [ClassDefinition(name='drug', singular_name=None, description='A substance intended for use in the diagnosis, cure, mitigation, treatment, or prevention of disease', note=None, comment='The CHEBI ID represents a role rather than a substance', examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['WD:Q12140', 'CHEBI:23888'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='chemical substance', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Drug.md) - A substance intended for use in the diagnosis, cure, mitigation, treatment, or prevention of disease
       * [ClassDefinition(name='organismal entity', singular_name=None, description='A named entity that is either a part of an organism, a whole organism, population or clade of organisms, excluding molecular entities', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['WD:Q7239'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biological entity', mixin=False, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](OrganismalEntity.md) - A named entity that is either a part of an organism, a whole organism, population or clade of organisms, excluding molecular entities
          * [ClassDefinition(name='biosample', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['biospecimen', 'sample'], mappings=['SIO:001050'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='organismal entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['in taxon'], slot_usage={}, apply_to=None, entity=False)](Biosample.md)
             * [ClassDefinition(name='cell line', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['CLO:0000031'], id_prefixes=['CLO'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='biosample', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](CellLine.md)
          * [ClassDefinition(name='individual organism', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:010000', 'WD:Q795052'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='organismal entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=[], subclass_of='NCBITaxon:1', values_from=[], symmetric=False, defining_slots=[], slots=['in taxon'], slot_usage={}, apply_to=None, entity=False)](IndividualOrganism.md)
             * [ClassDefinition(name='case', singular_name=None, description='An individual organism that has a patient role in some clinical context.', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['patient', 'proband'], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='individual organism', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Case.md) - An individual organism that has a patient role in some clinical context.
          * [ClassDefinition(name='life stage', singular_name=None, description='A stage of development or growth of an organism, including post-natal adult stages', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='organismal entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=[], subclass_of='UBERON:0000105', values_from=[], symmetric=False, defining_slots=[], slots=['in taxon'], slot_usage={}, apply_to=None, entity=False)](LifeStage.md) - A stage of development or growth of an organism, including post-natal adult stages
          * [ClassDefinition(name='anatomical entity', singular_name=None, description='A subcellular location, cell type or gross anatomical part', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:ANAT', 'SIO:010046', 'WD:Q4936952'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='organismal entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names=[], union_of=[], subclass_of='UBERON:0001062', values_from=[], symmetric=False, defining_slots=[], slots=['expresses', 'in taxon'], slot_usage={}, apply_to=None, entity=False)](AnatomicalEntity.md) - A subcellular location, cell type or gross anatomical part
             * [ClassDefinition(name='cell', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GO:0005623', 'CL:0000000', 'SIO:010001', 'WD:Q7868'], id_prefixes=['CL', 'PO'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='anatomical entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Cell.md)
             * [ClassDefinition(name='gross anatomical structure', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['tissue', 'organ'], mappings=['UBERON:0010000', 'SIO:010046', 'WD:Q4936952'], id_prefixes=['UBERON', 'PO', 'FAO'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='anatomical entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GrossAnatomicalStructure.md)
             * [ClassDefinition(name='cellular component', singular_name=None, description='A location in or around a cell', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['GO:0005575', 'SIO:001400', 'WD:Q5058355'], id_prefixes=['GO'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='anatomical entity', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](CellularComponent.md) - A location in or around a cell
          * [ClassDefinition(name='population of individual organisms', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['SIO:001061'], id_prefixes=['ANCESTRO'], in_subset=[], from_schema=None, alt_descriptions=[], is_a='organismal entity', mixin=False, mixins=['thing with taxon'], abstract=False, local_names={'ga4gh': 'population'}, union_of=[], subclass_of='PCO:0000001', values_from=[], symmetric=False, defining_slots=[], slots=['in taxon'], slot_usage={}, apply_to=None, entity=False)](PopulationOfIndividualOrganisms.md)
 * [ClassDefinition(name='occurrent', singular_name=None, description='A processual entity', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['BFO:0000003'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['regulates, process to process', 'has participant', 'has input', 'precedes'], slot_usage={}, apply_to=None, entity=False)](Occurrent.md) - A processual entity
    * [ClassDefinition(name='procedure', singular_name=None, description='A series of actions conducted in a certain order or manner', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:PROC'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='occurrent', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Procedure.md) - A series of actions conducted in a certain order or manner
    * [ClassDefinition(name='activity and behavior', singular_name=None, description='Activity or behavior of any independent integral living, organization or mechanical actor in the world', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:ACTI'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='occurrent', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](ActivityAndBehavior.md) - Activity or behavior of any independent integral living, organization or mechanical actor in the world
    * [ClassDefinition(name='phenomenon', singular_name=None, description='a fact or situation that is observed to exist or happen, especially one whose cause or explanation is in question', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UMLSSG:PHEN'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='occurrent', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](Phenomenon.md) - a fact or situation that is observed to exist or happen, especially one whose cause or explanation is in question
 * [ClassDefinition(name='ontology class', singular_name=None, description='a concept or class in an ontology, vocabulary or thesaurus', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['subclass of'], slot_usage={}, apply_to=None, entity=False)](OntologyClass.md) - a concept or class in an ontology, vocabulary or thesaurus
    * [ClassDefinition(name='gene ontology class', singular_name=None, description='an ontology class that describes a functional aspect of a gene, gene prodoct or complex', note=None, comment=None, examples=[], see_also=None, flags=['testing'], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='ontology class', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GeneOntologyClass.md) - an ontology class that describes a functional aspect of a gene, gene prodoct or complex
    * [ClassDefinition(name='organism taxon', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['WD:Q16521'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='ontology class', mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['NCBITaxon'], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](OrganismTaxon.md)
 * [ClassDefinition(name='relationship type', singular_name=None, description='An OWL property used as an edge label', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](RelationshipType.md) - An OWL property used as an edge label
### Mixins

 * [ClassDefinition(name='entity to disease association', singular_name=None, description='mixin class for any association whose object (target node) is a disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['object'], slots=['frequency qualifier', 'severity qualifier', 'onset qualifier', 'entity to disease association object'], slot_usage={'object': SlotDefinition(name='object', singular_name=None, description='disease', note=None, comment=None, examples=[Example(value='MONDO:0020066', description='Ehlers-Danlos syndrome')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='MONDO:0000001', values_from=[], symmetric=False, multivalued=False, domain=None, range='disease', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](EntityToDiseaseAssociation.md) - mixin class for any association whose object (target node) is a disease
 * [ClassDefinition(name='frequency quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['has count', 'has total', 'has quotient', 'has percentage'], slot_usage={}, apply_to=None, entity=False)](FrequencyQuantifier.md)
 * [ClassDefinition(name='gene grouping', singular_name=None, description='any grouping of multiple genes or gene products', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GeneGrouping.md) - any grouping of multiple genes or gene products
 * [ClassDefinition(name='gene ontology class', singular_name=None, description='an ontology class that describes a functional aspect of a gene, gene prodoct or complex', note=None, comment=None, examples=[], see_also=None, flags=['testing'], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='ontology class', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](GeneOntologyClass.md) - an ontology class that describes a functional aspect of a gene, gene prodoct or complex
 * [ClassDefinition(name='model to disease mixin', singular_name=None, description="This mixin is used for any association class for which the subject (source node) plays the role of a 'model', in that it recapitulates some features of the disease in a way that is useful for studying the disease outside a patient carrying the disease", note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['model to disease mixin subject', 'model to disease mixin relation'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='The entity that serves as the model of the disease. This may be an organism, a strain of organism, a genotype or variant that exhibits similar features, or a gene that when mutated exhibits features of the disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='The relationship to the disease\n  ', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range=None, required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of='model of', inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](ModelToDiseaseMixin.md) - This mixin is used for any association class for which the subject (source node) plays the role of a 'model', in that it recapitulates some features of the disease in a way that is useful for studying the disease outside a patient carrying the disease
 * [ClassDefinition(name='molecular interaction', singular_name=None, description='An interaction at the molecular level between two physical entities', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['molecular interaction subject', 'molecular interaction relation', 'molecular interaction object', 'molecular interaction interacting molecules category'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'relation': SlotDefinition(name='relation', singular_name=None, description='interaction relationship type', note=None, comment=None, examples=[Example(value='RO:0002447', description='the subject molecular phosphorylates the object molecule')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=['ro', 'mi'], symmetric=False, multivalued=False, domain=None, range='molecularly interacts with', required=False, inlined=False, primary_key=False, identifier=False, definitional=True, alias=None, path=None, subproperty_of='molecularly interacts with', inverse=None, is_class_field=False, role=None), 'object': SlotDefinition(name='object', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='molecular entity', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None), 'interacting molecules category': SlotDefinition(name='interacting molecules category', singular_name=None, description=None, note=None, comment=None, examples=[Example(value='MI:1048', description='smallmolecule-protein')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of='MI:1046', values_from=[], symmetric=False, multivalued=False, domain=None, range='ontology class', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](MolecularInteraction.md) - An interaction at the molecular level between two physical entities
 * [ClassDefinition(name='pathognomonicity quantifier', singular_name=None, description='A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='specificity quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PathognomonicityQuantifier.md) - A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease
 * [ClassDefinition(name='relationship quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](RelationshipQuantifier.md)
    * [ClassDefinition(name='frequency quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['has count', 'has total', 'has quotient', 'has percentage'], slot_usage={}, apply_to=None, entity=False)](FrequencyQuantifier.md)
    * [ClassDefinition(name='senstivity quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](SenstivityQuantifier.md)
    * [ClassDefinition(name='specificity quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](SpecificityQuantifier.md)
       * [ClassDefinition(name='pathognomonicity quantifier', singular_name=None, description='A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='specificity quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PathognomonicityQuantifier.md) - A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease
 * [ClassDefinition(name='senstivity quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](SenstivityQuantifier.md)
 * [ClassDefinition(name='specificity quantifier', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='relationship quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](SpecificityQuantifier.md)
    * [ClassDefinition(name='pathognomonicity quantifier', singular_name=None, description='A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='specificity quantifier', mixin=True, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=[], slot_usage={}, apply_to=None, entity=False)](PathognomonicityQuantifier.md) - A relationship quantifier between a variant or symptom and a disease, which is high when the presence of the feature implies the existence of the disease
 * [ClassDefinition(name='thing with taxon', singular_name=None, description='A mixin that can be used on any entity with a taxon', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=True, mixins=[], abstract=True, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=[], slots=['in taxon'], slot_usage={}, apply_to=None, entity=False)](ThingWithTaxon.md) - A mixin that can be used on any entity with a taxon
 * [ClassDefinition(name='variant to thing association', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a='association', mixin=True, mixins=[], abstract=True, local_names={'ga4gh': 'variant annotation'}, union_of=[], subclass_of=None, values_from=[], symmetric=False, defining_slots=['subject'], slots=['variant to thing association subject'], slot_usage={'subject': SlotDefinition(name='subject', singular_name=None, description='a sequence variant in which the allele state is associated with some other entity', note=None, comment=None, examples=[Example(value='ClinVar:38077', description='ClinVar representation of NM_000059.3(BRCA2):c.7007G>A (p.Arg2336His)'), Example(value='ClinGen:CA024716', description='chr13:g.32921033G>C (hg19) in ClinGen')], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], is_a=None, mixin=False, mixins=[], abstract=False, local_names=[], union_of=[], subclass_of=None, values_from=[], symmetric=False, multivalued=False, domain=None, range='sequence variant', required=False, inlined=False, primary_key=False, identifier=False, definitional=False, alias=None, path=None, subproperty_of=None, inverse=None, is_class_field=False, role=None)}, apply_to=None, entity=False)](VariantToThingAssociation.md)
### Slots

 * [association slot](association_slot.md) - any slot that relates an association to another entity
    * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [subject](subject.md) - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
    * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
       * [relation](relation.md) - the relationship type by which a subject is connected to an object in an association
    * [qualifiers](qualifiers.md) - connects an association to qualifiers that modify or qualify the meaning of that association
    * [provided by](provided_by.md) - connects an association to the agent (person, organization or group) that provided it
    * [sequence variant qualifier](sequence_variant_qualifier.md) - a qualifier used in an association where the variant
    * [has confidence level](has_confidence_level.md) - connects an association to a qualitative term denoting the level of confidence
    * [frequency qualifier](frequency_qualifier.md) - a qualifier used in a phenotypic association to state how frequent the phenotype is observed in the subject
    * [edge label](edge_label.md) - A high-level grouping for the relationship type. AKA minimal predicate. This is analogous to category for nodes.
    * [quantifier qualifier](quantifier_qualifier.md) - A measurable quantity for the object of the association
    * [publications](publications.md) - connects an association to publications supporting the association
    * [association type](association_type.md) - connects an association to the type of association (e.g. gene to phenotype)
    * [stage qualifier](stage_qualifier.md) - stage at which expression takes place
    * [onset qualifier](onset_qualifier.md) - a qualifier used in a phenotypic association to state when the phenotype appears is in the subject
    * [has evidence](has_evidence.md) - connects an association to an instance of supporting evidence
    * [clinical modifier qualifier](clinical_modifier_qualifier.md) - Used to characterize and specify the phenotypic abnormalities defined in the Phenotypic abnormality subontology, with respect to severity, laterality, age of onset, and other aspects
    * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
       * [object](object.md) - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.
    * [sex qualifier](sex_qualifier.md) - a qualifier used in a phenotypic association to state whether the association is specific to a particular sex.
    * [severity qualifier](severity_qualifier.md) - a qualifier used in a phenotypic association to state how severe the phenotype is in the subject
    * [negated](negated.md) - if set to true, then the association is negated i.e. is not true
 * [drug](drug.md)
 * [object](object.md) - disease
 * [quantifier qualifier](quantifier_qualifier.md) - can be used to indicate magnitude, or also ranking
 * [stage qualifier](stage_qualifier.md) - stage at which the gene is expressed in the site
 * [relation](relation.md) - The relationship to the disease
 * [subject](subject.md) - The entity that serves as the model of the disease. This may be an organism, a strain of organism, a genotype or variant that exhibits similar features, or a gene that when mutated exhibits features of the disease
 * [interacting molecules category](interacting_molecules_category.md)
 * [node property](node_property.md) - A grouping for any property that holds between a node and a value
    * [genome build](genome_build.md) - TODO
    * [has gene](has_gene.md) - connects and entity to a single gene
    * [iri](iri.md) *subsets*: (translator_minimal) - An IRI for the node. This is determined by the id using expansion rules.
    * [id](id.md) *subsets*: (translator_minimal) - A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI
       * [id](id.md) *subsets*: (translator_minimal) - A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI
    * [interbase coordinate](interbase_coordinate.md) - TODO
       * [end interbase coordinate](end_interbase_coordinate.md) - TODO
       * [start interbase coordinate](start_interbase_coordinate.md) - TODO
    * [latitude](latitude.md) - latitude
    * [update date](update_date.md) - date on which thing was updated. This can be applied to nodes or edges
    * [phase](phase.md) - TODO
    * [description](description.md) *subsets*: (translator_minimal) - a human-readable description of a thing
       * [description](description.md) *subsets*: (translator_minimal) - a human-readable description of a thing
    * [has chemical formula](has_chemical_formula.md) - description of chemical compound based on element symbols
    * [name](name.md) *subsets*: (translator_minimal) - A human-readable name for a thing
       * [name](name.md) *subsets*: (translator_minimal) - A human-readable name for a thing
    * [aggregate statistic](aggregate_statistic.md) - A grouping for any property that holds between a node and a value
       * [has quotient](has_quotient.md) - A grouping for any property that holds between a node and a value
       * [has count](has_count.md) - number of things with a particular property
       * [has percentage](has_percentage.md) - equivalent to has quotient multiplied by 100
       * [has total](has_total.md) - total number of things in a particular reference set
    * [longitude](longitude.md) - longitude
    * [filler](filler.md) - The value in a property-value tuple
    * [full name](full_name.md) - a long-form human readable name for a thing
    * [category](category.md) *subsets*: (translator_minimal) - Name of the high level ontology class in which this entity is categorized. Corresponds to the label for the biolink entity type class. In a neo4j database this MAY correspond to the neo4j label tag
    * [has biological sequence](has_biological_sequence.md) - connects a genomic feature to its sequence
       * [has biological sequence](has_biological_sequence.md) - connects a genomic feature to its sequence
    * [timepoint](timepoint.md) - a point in time
    * [systematic synonym](systematic_synonym.md) - more commonly used for gene symbols in yeast
    * [creation date](creation_date.md) - date on which thing was created. This can be applied to nodes or edges
    * [has zygosity](has_zygosity.md) - A grouping for any property that holds between a node and a value
 * [related to](related_to.md) - A grouping for any relationship type that holds between any two things
    * [located in](located_in.md) *subsets*: (translator_minimal) - holds between a material entity and a material entity or site within which it is located (but of which it is not considered a part)
    * [gene associated with condition](gene_associated_with_condition.md) *subsets*: (translator_minimal) - holds between a gene and a disease or phenotypic feature that the gene or its alleles/products may influence, contribute to, or correlate with
    * [interacts with](interacts_with.md) *subsets*: (translator_minimal) - holds between any two entities that directly or indirectly interact with each other
       * [genetically interacts with](genetically_interacts_with.md) *subsets*: (translator_minimal) - holds between two genes whose phenotypic effects are dependent on each other in some way - such that their combined phenotypic effects are the result of some interaction between the activity of their gene products. Examples include epistasis and synthetic lethality.
       * [physically interacts with](physically_interacts_with.md) *subsets*: (translator_minimal) - holds between two entities that make physical contact as part of some interaction
          * [molecularly interacts with](molecularly_interacts_with.md) *subsets*: (translator_minimal) - holds between two entities that make physical contact as part of some interaction
    * [has participant](has_participant.md) *subsets*: (translator_minimal) - holds between a process and a continuant, where the continuant is somehow involved in the process
       * [has input](has_input.md) *subsets*: (translator_minimal) - holds between a process and a continuant, where the continuant is an input into the process
    * [produces](produces.md) *subsets*: (translator_minimal) - holds between a material entity and a product that is generated through the intentional actions or functioning of the material entity
    * [subclass of](subclass_of.md) *subsets*: (translator_minimal) - holds between two classes where the domain class is a specialization of the range class
    * [in taxon](in_taxon.md) *subsets*: (translator_minimal) - connects a thing to a class representing a taxon
    * [derives into](derives_into.md) *subsets*: (translator_minimal) - holds between two distinct material entities, the old entity and the new entity, in which the new entity begins to exist when the old entity ceases to exist, and the new entity inherits the significant portion of the matter of the old entity
    * [precedes](precedes.md) *subsets*: (translator_minimal) - holds between two processes, where one completes before the other begins
    * [manifestation of](manifestation_of.md) *subsets*: (translator_minimal) - used in SemMedDB for linking things like dysfunctions and processes to some disease or syndrome
    * [has molecular consequence](has_molecular_consequence.md) - connects a sequence variant to a class describing the molecular consequence. E.g.  SO:0001583
    * [expresses](expresses.md) *subsets*: (translator_minimal) - holds between an anatomical entity and gene or gene product that is expressed there
    * [coexists with](coexists_with.md) *subsets*: (translator_minimal) - holds between two entities that are co-located in the same aggregate object, process, or spatio-temporal region
       * [in cell population with](in_cell_population_with.md) *subsets*: (translator_minimal) - holds between two genes or gene products that are expressed in the same cell type or population
       * [in pathway with](in_pathway_with.md) *subsets*: (translator_minimal) - holds between two genes or gene products that are part of in the same biological pathway
       * [co-localizes with](co-localizes_with.md) *subsets*: (translator_minimal) - holds between two entities that are observed to be located in the same place.
       * [in complex with](in_complex_with.md) *subsets*: (translator_minimal) - holds between two genes or gene products that are part of (or code for products that are part of) in the same macromolecular complex
    * [homologous to](homologous_to.md) *subsets*: (translator_minimal) - holds between two biological entities that have common evolutionary origin
       * [orthologous to](orthologous_to.md) *subsets*: (translator_minimal) - a homology relationship between entities (typically genes) that diverged after a speciation event.
       * [xenologous to](xenologous_to.md) *subsets*: (translator_minimal) - a homology relationship characterized by an interspecies (horizontal) transfer since the common ancestor.
       * [paralogous to](paralogous_to.md) *subsets*: (translator_minimal) - a homology relationship that holds between entities (typically genes) that diverged after a duplication event.
    * [contributes to](contributes_to.md) *subsets*: (translator_minimal) - holds between two entities where the occurrence, existence, or activity of one causes or contributes to the occurrence or generation of the other
       * [causes](causes.md) *subsets*: (translator_minimal) - holds between two entities where the occurrence, existence, or activity of one causes the occurrence or  generation of the other
    * [correlated with](correlated_with.md) *subsets*: (translator_minimal) - holds between a disease or phenotypic feature and a measurable molecular entity that is used as an indicator of the presence or state of the disease or feature.
       * [biomarker for](biomarker_for.md) *subsets*: (translator_minimal) - holds between a measurable molecular entity and a disease or phenotypic feature, where the entity is used as an indicator of the presence or state of the disease or feature.
       * [has biomarker](has_biomarker.md) *subsets*: (translator_minimal) - holds between a disease or phenotypic feature and a measurable molecular entity that is used as an indicator of the presence or state of the disease or feature.
    * [model of](model_of.md) *subsets*: (translator_minimal) - holds between an entity and some other entity it approximates for purposes of scientific study, in virtue of its exibiting similar features of the studied entity.
    * [overlaps](overlaps.md) *subsets*: (translator_minimal) - holds between entties that overlap in their extents (materials or processes)
       * [part of](part_of.md) *subsets*: (translator_minimal) - holds between parts and wholes (material entities or processes)
       * [has part](has_part.md) *subsets*: (translator_minimal) - holds between wholes and their parts (material entities or processes)
    * [treated by](treated_by.md) *subsets*: (translator_minimal) - holds between a disease or phenotypic feature and a therapeutic process or chemical substance that is used to treat the condition
    * [derives from](derives_from.md) *subsets*: (translator_minimal) - holds between two distinct material entities, the new entity and the old entity, in which the new entity begins to exist when the old entity ceases to exist, and the new entity inherits the significant portion of the matter of the old entity
    * [affects risk for](affects_risk_for.md) *subsets*: (translator_minimal) - holds between two entities where exposure to one entity alters the chance of developing the other
       * [predisposes](predisposes.md) *subsets*: (translator_minimal) - holds between two entities where exposure to one entity increases the chance of developing the other
       * [prevents](prevents.md) *subsets*: (translator_minimal) - holds between an entity whose application or use reduces the likelihood of a potential outcome.  Typically used to associate a chemical substance, exposure, activity, or medical intervention that can prevent the onset a disease or phenotypic feature.
    * [expressed in](expressed_in.md) *subsets*: (translator_minimal) - holds between a gene or gene product and an anatomical entity in which it is expressed
    * [participates in](participates_in.md) *subsets*: (translator_minimal) - holds between a continuant and a process, where the continuant is somehow involved in the process
       * [actively involved in](actively_involved_in.md) *subsets*: (translator_minimal) - holds between a continuant and a process or function, where the continuant actively contributes to part or all of the process or function it realizes
          * [capable of](capable_of.md) *subsets*: (translator_minimal) - holds between a continuant and process or function, where the continuant alone has the ability to carry out the process or function.
    * [occurs in](occurs_in.md) *subsets*: (translator_minimal) - holds between a process and a material entity or site within which the process occurs
    * [has phenotype](has_phenotype.md) *subsets*: (translator_minimal) - holds between a biological entity and a phenotype, where a phenotype is construed broadly as any kind of quality of an organism part, a collection of these qualities, or a change in quality or qualities (e.g. abnormally increased temperature).
    * [has gene product](has_gene_product.md) *subsets*: (translator_minimal) - holds between a gene and a transcribed and/or translated product generated from it
    * [location of](location_of.md) *subsets*: (translator_minimal) - holds between material entity or site and a material entity that is located within it (but not considered a part of it)
    * [affects](affects.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
       * [regulates](regulates.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
          * [regulates, entity to entity](regulates_entity_to_entity.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
             * [negatively regulates, entity to entity](negatively_regulates_entity_to_entity.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
             * [positively regulates, entity to entity](positively_regulates_entity_to_entity.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
          * [regulates, process to process](regulates_process_to_process.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
             * [positively regulates, process to process](positively_regulates_process_to_process.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
             * [negatively regulates, process to process](negatively_regulates_process_to_process.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
          * [positively regulates](positively_regulates.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
          * [negatively regulates](negatively_regulates.md) *subsets*: (translator_minimal) - describes an entity that has a direct affect on the state or quality of another existing entity. Use of the 'affects' predicate implies that the affected entity already exists, unlike predicates such as 'affects risk for' and 'prevents, where the outcome is something that may or may not come to be.
       * [disrupts](disrupts.md) *subsets*: (translator_minimal) - describes a relationship where one entity degrades or interferes with the structure, function, or occurrence of another.
       * [treats](treats.md) *subsets*: (translator_minimal) - holds between a therapeutic procedure or chemical substance and a disease or phenotypic feature that it is used to treat
    * [same as](same_as.md) *subsets*: (translator_minimal) - holds between two entities that are considered equivalent to each other
 * [has gene](has_gene.md) - Each allele can be associated with any number of genes
 * [has exposure parts](has_exposure_parts.md)
 * [treats](treats.md)
 * [has count](has_count.md) - number in object population that carry a particular allele, aka allele count
 * [has quotient](has_quotient.md) - frequency of allele in population, expressed as a number with allele divided by number in reference population, aka allele frequency
 * [has total](has_total.md) - number all populations that carry a particular allele, aka allele number
### Types

#### Built in

 * **boolean**
 * **date**
 * **double**
 * **float**
 * **integer**
 * **string**
 * **time**
#### Defined

 * [TypeDefinition(name='biological sequence', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](BiologicalSequence.md) (**string**)
 * [TypeDefinition(name='chemical formula type', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](ChemicalFormulaType.md) (**string**)
 * [TypeDefinition(name='chemical formula value', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof=None)](ChemicalFormulaValue.md) (**string**)
 * [TypeDefinition(name='evidence instance', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof=None)](EvidenceInstance.md) (**string**)
 * [TypeDefinition(name='frequency value', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UO:0000105'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof=None)](FrequencyValue.md) (**string**)
 * [TypeDefinition(name='identifier type', singular_name=None, description='A string that is intended to uniquely identify a thing May be URI in full or compact (CURIE) form', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](IdentifierType.md) (**string**) - A string that is intended to uniquely identify a thing May be URI in full or compact (CURIE) form
 * [TypeDefinition(name='iri type', singular_name=None, description='An IRI', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](IriType.md) (**string**) - An IRI
 * [TypeDefinition(name='label type', singular_name=None, description='A string that provides a human-readable name for a thing', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](LabelType.md) (**string**) - A string that provides a human-readable name for a thing
 * [TypeDefinition(name='narrative text', singular_name=None, description='A string that provides a human-readable description of something', note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](NarrativeText.md) (**string**) - A string that provides a human-readable description of something
 * [TypeDefinition(name='perecentage frequency value', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UO:0000187'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='double')](PerecentageFrequencyValue.md) (**double**)
 * [TypeDefinition(name='phenotype', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof=None)](Phenotype.md) (**string**)
 * [TypeDefinition(name='quotient', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=['ratio'], mappings=['UO:0010006'], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='double')](Quotient.md) (**double**)
 * [TypeDefinition(name='symbol type', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](SymbolType.md) (**string**)
 * [TypeDefinition(name='time type', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=[], id_prefixes=[], in_subset=[], from_schema=None, alt_descriptions=[], typeof='time')](TimeType.md) (**time**)
 * [TypeDefinition(name='unit', singular_name=None, description=None, note=None, comment=None, examples=[], see_also=None, flags=[], prefixes=[], aliases=[], mappings=['UO:0000000'], id_prefixes=['UO'], in_subset=[], from_schema=None, alt_descriptions=[], typeof='string')](Unit.md) (**string**)
